src/Generator.h

```diff
@@ -842,29 +881,137 @@ decltype(!(T)0) operator!(const GeneratorParam<T> &a) { return !(T)a; }
 
 namespace Internal {
 
-enum class IOKind { Scalar, Function };
+template<typename T2> class GeneratorInput_Buffer;
+
+enum class IOKind { Scalar, Function, Buffer };
+
+/**
+ * StubInputBuffer is the placeholder that a Stub uses when it requires 
+ * a Buffer for an input (rather than merely a Func or Expr). It is constructed
+ * to allow only two possible sorts of input:
+ * -- Assignment of an Input<Buffer<>>, with compatible type and dimensions;
+ * this is useful in both AOT and JIT compilation modes.
+ * -- Assignment of a Buffer<>, with compatible type and dimensions;
```

Comment:
Per discussion on the top-level description text, I think it should be useful and valid to pass a constant image for the AOT case as well and I expect it might work correctly. (The constant image will be compiled into the generated code as constant data. Making it an Input<Buffer<>> allows adding constraints on the layout for efficiency.) If this does work, please update this comment.